src/barril/units/_quantity.py

```diff
@@ -677,9 +678,17 @@ def Convert(self, value, to_unit):
         :returns:
             An object with values to the passed unit.
         """
-        return self._unit_database.Convert(
-            self._composing_categories, self._composing_units, to_unit, value
-        )
+        try:
+            return self._unit_database.Convert(
+                self._composing_categories, self._composing_units, to_unit, value
+            )
+        except ComposedUnitError:
+            return self._unit_database.Convert(
+                self._category,
+                self._CreateUnitsWithJoinedExponentsString(),
```

Comment:
Isn't `self._CreateUnitsWithJoinedExponentsString()` already cached in `self._Units`?